lib/src/facts/linux/networking_resolver.cc

```diff
@@ -71,4 +89,85 @@ namespace facter { namespace facts { namespace linux {
         return interface;
     }
 
+    void networking_resolver::read_routing_table()
+    {
+        auto ip_command = lth_exe::which("ip");
+        if (ip_command.empty()) {
+            LOG_DEBUG("Could not find the 'ip' command. Network bindings will not be populated from routing table");
+            return;
+        }
+
+        auto parse_route_line = [](string& line, std::vector<route>& routes) {
+            vector<boost::iterator_range<string::iterator>> parts;
+            boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+            if (parts.size() % 2 != 1) {
+                LOG_WARNING("Could not process routing table entry: Expected a destination followed by key/value pairs, got '{}'", line);
+                return true;
+            }
+
+            route r;
+            r.destination.assign(parts[0].begin(), parts[0].end());
```

Comment:
parts isn't a vector of strings, it's a vector of iterator_ranges which point directly into the original string. If I wanted to move here I'd have to use the version of `split` that creates copies of every field, which would include the values for the key/value pairs I'm not using. It would maybe simplify the code, but since I know there's at least a couple standard route fields I'm ignoring (via, proto, and meetric), this seemed "more right".